lib/grape/error_formatter/base.rb

```diff
@@ -3,9 +3,13 @@ module ErrorFormatter
     module Base
       def present(message, env)
         present_options = {}
-        present_options[:with] = message.delete(:with) if message.is_a?(Hash)
+        presented_message = message
+        if presented_message.is_a?(Hash) && presented_message.key?(:with)
+          present_options[:with] = presented_message[:with]
+          presented_message = presented_message.reject { |k, _| k == :with }
+        end
```

Comment:
I did it this way to avoid creating the extra hash when it is unnecessary. But I think you're right, since the argument here may also be the result of this method, it's probably better to always dupe.